src/gasPriceOracle/adapters/ethereum.ts

```diff
@@ -1,28 +1,19 @@
-import { providers } from "ethers";
-import { BigNumber, bnZero } from "../../utils";
-import { GasPriceEstimate } from "../types";
-import { gasPriceError } from "../util";
+import { PublicClient } from "viem";
+import { InternalGasPriceEstimate } from "../types";
 
-export async function eip1559(provider: providers.Provider, chainId: number): Promise<GasPriceEstimate> {
-  const feeData = await provider.getFeeData();
-
-  [feeData.lastBaseFeePerGas, feeData.maxPriorityFeePerGas].forEach((field: BigNumber | null) => {
-    if (!BigNumber.isBigNumber(field) || field.lt(bnZero)) gasPriceError("getFeeData()", chainId, feeData);
-  });
-
-  const maxPriorityFeePerGas = feeData.maxPriorityFeePerGas as BigNumber;
-  const maxFeePerGas = maxPriorityFeePerGas.add(feeData.lastBaseFeePerGas as BigNumber);
-
-  return { maxPriorityFeePerGas, maxFeePerGas };
+export function eip1559(provider: PublicClient, _chainId: number): Promise<InternalGasPriceEstimate> {
+  return provider.estimateFeesPerGas();
```

Comment:
The linter is erroring here. But I assume the reason why we keep the same function signature is for backwards-compatibility? In that case, we can add something like 
```
{ "argsIgnorePattern": "^_" }
```
to eslint config. See https://eslint.org/docs/latest/rules/no-unused-vars#argsignorepattern